### PR TITLE
retrive retitle if no good title provided

### PR DESCRIPTION
**Pull request by shohoku11wrj**
image

### CODE REVIEW

File: haoshiyou/src/loopbacksdk/models/HsyListing.ts [hunk 1]
```diff
@@ -81,6 +81,61 @@ export class HsyListing implements HsyListingInterface {
...
+   * 2: 地点
+   * 3: 房间
+   * 4: available date
+   */
+  private isGoodTitle(): boolean{
+    let targets = ["出租","招租","求租","转租","合租","长租","短租","找室友","招室友"];
+    let includeTarget = this.includesValue(this.title, targets);
+    let cities = ["san","santa","francisco","berkeley","oakland","mateo","palo alto","daly","forster","redwood","menlo park","stanford","mountain view","sunnyvale","cupertino","jose","santa clara","campbell","milpitas","fremont","newark","union","hayward","东湾","南湾","三番","旧金山","区","半岛"];
+    let includeLocation = this.includesValue(this.title.toLowerCase(), cities);
+    let rooms = ["主卧","次卧","单房","两房","1B1B","2B1B","2B2B","4B","3B","2B","1B","猫","狗"];
```
Comment (xinbenlv) [inline haoshiyou/src/loopbacksdk/models/HsyListing.ts]: Can we try to not re-declare the same thing twice? 
```
let rooms = ["主卧","次卧","单房","两房","1B1B","2B1B","2B2B","4B","3B","2B","1B","猫","狗"];
```
here and below.

And, as a product decision, let's keep "猫", "狗" separate?

File: haoshiyou/src/loopbacksdk/models/HsyListing.ts [hunk 2]
```diff
@@ -81,6 +81,61 @@ export class HsyListing implements HsyListingInterface {
...
+  }
+  /**
+   * good title should contain following information
+   * 1: 出租、求租
+   * 2: 地点
+   * 3: 房间
+   * 4: available date
+   */
+  private isGoodTitle(): boolean{
+    let targets = ["出租","招租","求租","转租","合租","长租","短租","找室友","招室友"];
```
Comment (xinbenlv) [inline haoshiyou/src/loopbacksdk/models/HsyListing.ts]: this is also declared twice?

File: haoshiyou/src/loopbacksdk/models/HsyListing.ts [hunk 3]
```diff
@@ -81,6 +81,61 @@ export class HsyListing implements HsyListingInterface {
   constructor(data?: HsyListingInterface) {
     Object.assign(this, data);
   }
+  /**
+   * regenerate title
+   */
+  private getReTitle(): string{
```
Comment (xinbenlv) [inline haoshiyou/src/loopbacksdk/models/HsyListing.ts]: make the function name more meaningful?

File: haoshiyou/src/loopbacksdk/models/HsyListing.ts [hunk 4]
```diff
@@ -81,6 +81,61 @@ export class HsyListing implements HsyListingInterface {
...
+    let rooms = ["主卧","次卧","单房","两房","1B1B","2B1B","2B2B","4B","3B","2B","1B","猫","狗"];
+    return "[" + this.getValue(this.content, targets, "信息") + "]"
+        + " " + this.getValue(this.content, rooms, "有一房间")
+        + " @ " + (this.addressCity ? this.addressCity.charAt(0).toUpperCase() + this.addressCity.slice(1) : "湾区")
+        + (this.addressZipcode ? ", " + this.addressZipcode : "");
+  }
+  /**
+   * retrieve any matched values in content
+   */
+  private getValue(content:string, values:Array<any>, defaultValue:string): string{
```
Comment (xinbenlv) [inline haoshiyou/src/loopbacksdk/models/HsyListing.ts]: Includes and getValue can be rewrite into 

let matcher = new RegEx(targets.join('|')) and just it directly?

File: haoshiyou/src/loopbacksdk/models/HsyListing.ts [hunk 5]
```diff
@@ -81,6 +81,61 @@ export class HsyListing implements HsyListingInterface {
...
+   * good title should contain following information
+   * 1: 出租、求租
+   * 2: 地点
+   * 3: 房间
+   * 4: available date
+   */
+  private isGoodTitle(): boolean{
+    let targets = ["出租","招租","求租","转租","合租","长租","短租","找室友","招室友"];
+    let includeTarget = this.includesValue(this.title, targets);
+    let cities = ["san","santa","francisco","berkeley","oakland","mateo","palo alto","daly","forster","redwood","menlo park","stanford","mountain view","sunnyvale","cupertino","jose","santa clara","campbell","milpitas","fremont","newark","union","hayward","东湾","南湾","三番","旧金山","区","半岛"];
```
[the remark this answers is not in the document]
Comment (shohoku11wrj) [reply haoshiyou/src/loopbacksdk/models/HsyListing.ts]: moved to haoshiyou/src/pages/listings-tab/listing-ux-item.comp.ts

File: haoshiyou/src/loopbacksdk/models/HsyListing.ts (hunk 4 above)
Comment (shohoku11wrj) [reply haoshiyou/src/loopbacksdk/models/HsyListing.ts]: change to "getFirstMatch()".

**Comment by shohoku11wrj** [comment]
change to new pull request https://github.com/haoshiyou/haoshiyou-client/pull/97

**Pull request closed without merging**